src/mark.js

```diff
@@ -7,9 +7,10 @@ const TypedArray = Object.getPrototypeOf(Uint8Array);
 const objectToString = Object.prototype.toString;
 
 export class Mark {
-  constructor(data, channels = [], options = {}) {
+  constructor(data, channels = [], {facet, ...options} = {}) {
     const names = new Set();
     this.data = data;
+    this.facet = facet;
```

Comment:
We should do some validation and coercion here.

I also think we can unify the types a little more. Currently:

* `"exclude"` is a string,
* `null` and `undefined` are equivalent, and
* `true` and `false` are booleans.

If we want to be more analogous to the *scale*.axis option, perhaps:

* `"exclude"`
* `"include"`, and `true` is shorthand for `"include"`
* `undefined` is `"include"` if data is strict equal, and otherwise `null` to disable faceting
* `null` or any other falsey value is mapped to `null` to disable faceting

Any other truthy value, after being coerced to a string, would throw an error. We’d use the `keyword` helper:

https://github.com/observablehq/plot/blob/068f8d8125a910ee1a528c8b299a79c27ca1a9a5/src/mark.js#L126